src/main/kotlin/org/rust/lang/core/psi/ext/RsPatFieldFull.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.psi.ext
+
+import com.intellij.lang.ASTNode
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RsPatFieldFull
+import org.rust.lang.core.psi.RsPatStruct
+import org.rust.lang.core.resolve.collectResolveVariants
+import org.rust.lang.core.resolve.processStructPatternFieldResolveVariants
+import org.rust.lang.core.resolve.ref.ResolveCacheDependency
+import org.rust.lang.core.resolve.ref.RsReference
+import org.rust.lang.core.resolve.ref.RsReferenceCached
+
+
+val RsPatFieldFull.parentStructPattern: RsPatStruct
+    get() = this.ancestors.filterIsInstance<RsPatStruct>().first()
```

Comment:
`this` is redundant here. And I think this can be implemented using `ancestorStrict` (see `RsFieldDecl.parentStruct`).